support/util/expose.go

```diff
@@ -0,0 +1,21 @@
+package util
+
+import (
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+)
+
+func ServicePublishingStrategyByTypeForHCP(hcp *hyperv1.HostedControlPlane, svcType hyperv1.ServiceType) *hyperv1.ServicePublishingStrategy {
+	for _, mapping := range hcp.Spec.Services {
+		if mapping.Service == svcType {
+			return &mapping.ServicePublishingStrategy
+		}
+	}
+	return nil
+}
+
+func HasPublicLoadBalancerForPrivateRouter(hcp *hyperv1.HostedControlPlane) bool {
```

Comment:
Does it make sense to use this here as well?
https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L583-L589

Reply:
Done